### PR TITLE
Add '-' as accepted username character

### PR DESCRIPTION
**Pull request by diamondo25**
Someone from iOS legitly tried to join with this character in her name. Lets accept it.
Android doesn't seem to accept it, but other characters are (like euro sign). Needs some investigation.

### CODE REVIEW

**Comment by Muqsit** [comment]
Noooo.... This feature is right out of the blue. Even bukkit doesn't have it. Negative.

**Comment by NycuRO** [comment]
Stupid ideea,please close :/

**Comment by diamondo25** [comment]
> Noooo.... This feature is right out of the blue. Even bukkit doesn't have it. Negative.
This is more something for PE than the regular version of minecraft, so of course bukkit didn't have this (actually, the mojang account page restricts the use of characters in your player name)

File: src/main/java/cn/nukkit/Player.java [hunk 1]
```diff
@@ -1794,7 +1794,7 @@ public void handleDataPacket(DataPacket packet) {
                         if ((c >= 'a' && c <= 'z') ||
                                 (c >= 'A' && c <= 'Z') ||
```
Comment (TheDiamondYT1) [inline src/main/java/cn/nukkit/Player.java]: For the Nukkit devs (sorry for spam):

Why not use regex instead?

**Comment by IrishUHC** [comment]
I support this change, - is an acceptable character in my opinion. Just because Bukkit doesn't have it, does it mean Nukkit can't?

**Comment by TheDiamondYT1** [comment]
@IrishUHC I agreez

**Comment by Snake1999** [comment]
As far as I know, many nukkit servers uses databases as well, and uses '-' to replace blanks in user names in convenience. If - is accepted to legal usernames, then we will have a big conflict problem. 

That's also why windows 10 not 9 after windows 8 - many softwares already detect number 9 for windows 95 98 and so on.

**Comment by diamondo25** [comment]
Because other software that depends on us is written badly, we cannot do this? Thats a weird dependency. Can you give me an example plugi that does this?

**Pull request closed without merging**